artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/ProtonServerSenderContext.java

```diff
@@ -1115,8 +1116,15 @@ public Consumer init(ProtonServerSenderContext senderContext) throws Exception {
             } else {
                // if not we look up the address
                AddressQueryResult addressQueryResult = null;
+
+               // Set this to the broker configured default for the address prior to the lookup so that
+               // an auto create will actually use the configured defaults.  The actual query result will
+               // contain the true answer on what routing type the address actually has though.
+               routingTypeToUse = sessionSPI.getDefaultRoutingType(addressToUse);
+               routingTypeToUse = routingTypeToUse == null ? ActiveMQDefaultConfiguration.getDefaultRoutingType() : routingTypeToUse;
```

Comment:
I think this one might be more obvious with a simple if doing a potential update, rather than using the ternary and setting itself to itself in many/most cases.

Reply:
Agree to disagree